src/tscore/ink_file.cc

```diff
@@ -42,6 +42,21 @@ using ioctl_arg_t = union {
   off_t    off;
 };
 
+/** Open and possibly create a file.
+ *
+ * This is a wrapper for the POSIX open() call that will retry the call if a
+ * transient error occurs.
+ */
+int
+safe_open(char const *path, int oflag, mode_t mode)
+{
+  int r{-1};
+  do {
+    r = open(path, oflag, mode);
+  } while ((r < 0) && (errno == EINTR));
+  return r;
+}
+
```

Comment:
I'm not sure why this was originally inlined in the header.  But remember that functions declared with inline are not necessarily inlined.  If the compiler considers it non-optimal to actually inline, if will put the function in a vague linkage section, which prevents multiple copies of it from being present in the linked executable:  https://gcc.gnu.org/onlinedocs/gcc/Vague-Linkage.html .

Reply:
I think we wouldn't usually put definitions in the header file unless we think they may be useful to inline. Should we keep this in the header file with the `inline` declaration, then?

Reply:
Personally I'm fine with it not being inline.